### ISSUE TITLE
[Flow] Allow `number` as route param

### ISSUE THREAD

**Issue by Andreyco** [react-community/react-navigation#2062]
### Current Behavior
Flow complains about setting params other that `string` type.

Failing use cases:
```js
// at call site
navigation.navigate('UserProfile', { userId: 1 })

// Navigator config
{
  initialRouteName: 'UserProfile',
  initiarRouteParams: {
    userId: 1,
  },
}
```
and probably others.

When reading those params, it is often necessary to convert params back to appropriate type, e.g `string` to `number`.

### Expected Behavior
Being able to pass `number`, or any other serialisable data type as route param.

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation |  1.0.0-beta.11
| react-native     |  *
| node             | *
| npm or yarn      | *

### Notes
I am not sure, whether params of type other that `string` type will work, or break anything. Maybe it feasible to adjust type definitions.

**Comment by Andreyco** [comment]
I think this is fixed via #1574

**Comment by Andreyco** [comment]
I think this is closed via #1574
